### PR TITLE
Bugfix: latest uploads fail

### PR DESCRIPTION
**Pull request by lsh-0**
fixes a bug where >1 backups of a single target (in this case, multiple dbs on mysql) fail to find the latest backup when asking for one of the databases specifically (it will return the other db).

* fixes the bug - some sloppy code
* added ability to clean 'directories' of s3 files after a test run
* removed some hardcoded references to `/tmp`

### CODE REVIEW

File: ubr/conf.py [hunk 1]
```diff
@@ -127,5 +127,5 @@ def cfg(path, default=0xDEADBEEF):
     # https://www.postgresql.org/docs/9.2/static/libpq-pgpass.html
     #'pass':
     'host': cfg('postgresql.host', 'localhost'),
-    'port': cfg('postgresql.port', 5432),
+    'port': int(cfg('postgresql.port', 5432)),
```
Comment (giorgiosironi) [inline ubr/conf.py]: I tried running the tests locally and this should fix it

File: ubr/s3.py [hunk 2]
```diff
@@ -99,8 +99,14 @@ def parse_s3_project_files(bucket, project):
     return parse_path_list(s3_project_files(bucket, project))[project]
 """
 
-def filterasf(file_list, project, host, filename):
-    import re
+def filterasf(file_list, project, host, target=None, filename=''):
```
Comment (giorgiosironi) [inline ubr/s3.py]: a name more understandable than `asf`? https://en.wiktionary.org/wiki/asf

File: ubr/conf.py (hunk 1 above)
Comment (lsh-0) [inline ubr/conf.py]: yes, I was surprised I hadn't encountered this yet. it's when you have `postgresql.port` explicitly defined in `app.cfg` that it's read in as a string. if it's not there, it uses the integer. I do the same with mysql a few lines up.

File: ubr/s3.py (hunk 2 above)
Comment (lsh-0) [inline ubr/s3.py]: `filter-as-f` - I'm mystified what `f` originally was. perhaps it *was* short for `filter-asfasdf` , code archaeologists will be pondering this for years to come. I'll give it a new name.

**Comment by lsh-0** [comment]
all fixed up for you. I noticed I don't have any coverage testing, but it's up to 87% which is nice. I'm trying to get most of the python projects up to 80-90%.

**Pull request merged**